Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/index/type/BaseType.java

```diff
@@ -39,14 +45,104 @@
     }
 
     /**
+     * Method for adding relationship between some object and list of process
+     * objects.
+     * 
+     * @param processes
+     *            list
+     * @return JSONArray
+     */
+    @SuppressWarnings("unchecked")
+    JSONArray addProcessRelation(List<Process> processes) {
+        JSONArray jsonArray = new JSONArray();
+        if (processes != null) {
+            for (Process process : processes) {
+                jsonArray.add(addIdForRelation(process.getId()));
+            }
+        }
+        return jsonArray;
+    }
+
+    /**
+     * Method for adding relationship between some object and list of property
+     * objects.
+     * 
+     * @param properties
+     *            list
+     * @return JSONArray
+     */
+    @SuppressWarnings("unchecked")
+    JSONArray addPropertyRelation(List<Property> properties) {
+        JSONArray jsonArray = new JSONArray();
+        if (properties != null) {
+            for (Property property : properties) {
+                jsonArray.add(addIdForRelation(property.getId()));
+            }
+        }
+        return jsonArray;
+    }
+
+    /**
+     * Method for adding relationship between some object and list of user
+     * objects.
+     * 
+     * @param users
+     *            list
+     * @return JSONArray
+     */
+    @SuppressWarnings("unchecked")
+    JSONArray addUserRelation(List<User> users) {
+        JSONArray jsonArray = new JSONArray();
+        if (users != null) {
+            for (User user : users) {
+                jsonArray.add(addIdForRelation(user.getId()));
+            }
+        }
+        return jsonArray;
+    }
+
+    /**
+     * Method for adding relationship between some object and list of user group
+     * objects.
+     *
+     * @param userGroups
+     *            list
+     * @return JSONArray
+     */
+    @SuppressWarnings("unchecked")
+    JSONArray addUserGroupRelation(List<UserGroup> userGroups) {
+        JSONArray jsonArray = new JSONArray();
+        if (userGroups != null) {
+            for (UserGroup userGroup : userGroups) {
+                jsonArray.add(addIdForRelation(userGroup.getId()));
+            }
+        }
+        return jsonArray;
```

Comment:
I think you can unify the previous four methods with a generic. Didn't they all extend BaseType?
What exactly do they do? What is the "some object" you talk about in the javadoc?